### DISCUSSION TITLE
Combine rangeStrategy bump + update-lock

### DISCUSSION THREAD

**Discussion by RolphH** [renovatebot/renovate#13268] [Request Help] [answered]
In our projects we would like to use the rangeStrategy `bump` to bump the minor/patch version in our `composer.json` files. At the same time, we would like to have the `composer.lock` being updated, even for those packages that don't bump version. 

For example, assume following:

```
composer.json
{
   "require": {
     "foo": "^1.0",
     "bar": "^2.1"
}
```
``` 
composer.lock
{
   "packages": [{
      "name": "foo",
      "version": "v1.0.0"
   },
   {
      "name": "bar",
      "version": "v2.1.0"
    }]
}
```
Now assume that for `foo` there is minor update version `1.1` available, but for `bar` there is only a patch version available `v2.1.1`.
With `bump`, only the composer.json will be bumped (and the lock file with it) for `foo`, but `bar` won't be touched. With `update-lockfile`, only the composer.lock is touched and `bar` will be locked to `v2.1.1`.

Is this possible by specifying packageRules?

**Comment by viceice** [comment]
yes, you can configure it via package rules

**Comment by rarkins** [comment, marked as answer]
I think you should convert to using x.y.z precision in `composer.json` so that you can reliably bump each time. What do you gain by keeping as `x.y`, considering that (a) it's locked, and (b) you want an update from Renovate every time anyway? You don't save any pull requests by having short precision, but you lose visibility into `composer.json` because you need to dig into `composer.lock` to know which exact version you're using.

**Reply by RolphH** [reply]
Switching to x.y.z indeed fixes what I needed. Wasn't aware that the behaviour was slightly different between x.y and x.y.z. Using x.y.z and bump strategy works perfect. Thanks
